internal/pkg/service/buffer/storage/compression/reader/reader.go

```diff
@@ -12,10 +12,10 @@ import (
 )
 
 // New wraps the specified reader with the compression reader.
-func New(r io.Reader, cfg compression.Config) (io.Reader, error) {
+func New(r io.Reader, cfg compression.Config) (io.ReadCloser, error) {
 	switch cfg.Type {
```

Comment:
Upravil som return type pre compression new reader/writer.

`writechain` a `readchain` je to jedno, tam sa automaticky detekuje, ci ma dany writer/reader `Close` metodu.

Ale v inych testoch, kde som potreboval zavolat close manualne mi to robilo problem.

Co sa tyka logiky, tak je rozdiel, ze pri `TypeNone` sa nevrati rovnaky `reader` ale obali sa do `NopCloser`.